frontend/src/App.tsx

```diff
@@ -428,6 +430,7 @@ export class App extends PureComponent<Props, State> {
 
     this.setState({
       sharingEnabled: Boolean(config.sharingEnabled),
```

Comment:
In a world where allowRunOnSave is a longer-term, public feature, it might be worth bundling these two together (and adding any other config options that affect the menu, or otherwise need to be transferred in some fashion.)

We're not in that world, so I think this is fine; just noting that generally it's a code smell when we end up threading a new parameter through a bunch of functions and classes. It usually means those signatures could be consolidated somehow